linux/system/repo.sls

```diff
@@ -91,6 +91,16 @@ linux_repo_{{ name }}_key:
 
 {%- elif repo.key_url|default(False) %}
 
+{%- if system.proxy.get('pkg', {}).get('enabled', False) and system.proxy.get('pkg', {}).get('http', None) %}
+
+linux_repo_{{ name }}_key:
+  cmd.wait:
+    - name: "curl --proxy {{ system.proxy.get('pkg', {}).get('http', None) }}  -s {{ repo.key_url }} | apt-key add -"
```

Comment:
I know, but the system.proxy.pkg was actually assumed to be apt cachier rather than http/s proxy. You were expected to set up system wide http/s proxy by linux:system:proxy:https or linux:system:proxy:http.

Sure README needs an update